lib/delegate/proxy.js

```diff
@@ -82,7 +82,29 @@ var proto = {
                         registry: Url.parse(registry).host,
                         path: request.path,
                         statusCode: res ? res.statusCode : -1,
-                        id: request.id
+                        id: request.id,
+                        tags: 'proxy',
+                        data: {}
+                    });
+
+                    respondStart = Date.now();
+                    request.raw.res.socket.once('timeout', function () {
```

Comment:
We know the connection is [handled correctly](http://nodejs.org/api/net.html#net_event_timeout) (e.g. destroyed) in all code paths even though we're listening for timeouts here?